### ISSUE TITLE
Highcharts Bug: Unable to rely on `pointDescriptionEnabledThreshold` to ensure screen reader access

### ISSUE THREAD

**Issue by khiga8** [highcharts/highcharts#22149]
Fiddle: https://jsfiddle.net/j4yb1etz/

When a chart renders more than 200 data points, a screen reader user doesn't have the ability to interact with the chart directly and hear the datapoint announced.

While there is a setting to set [pointDescriptionEnabledThreshold](https://api.highcharts.com/highcharts/accessibility.series.pointDescriptionEnabledThreshold) to `false` so that the datapoints are always exposed to screen reader users, we're observing significant performance issues, especially because many of our charts render hundreds to thousands of datapoints.

The more points we render, the more `<path role="img" aria-label"` elements we render on the page which seems to significantly impact performance.

We would like screen reader users to always have the option to interact directly with the chart without the performance concerns. Do you have a recommendation for how to get around this performance issue? 

Since a sighted user/sighted keyboard user can navigate the points using a keyboard no matter the number of datapoints, we feel like this should also be an option for screen reader users. While there's an option to access the data via table or CSV, this doesn't feel like the same experience, and it feels confusing that there are some charts screen reader users can interact with, and some charts where they cannot.

If it's not possible to performantly expose these datapoints to screen reader users with the existing approach, one option we're thinking of exploring is to use a live region announcement to announce the datapoint. Though the points aren't exposed to screen reader users, one does still have the ability to keyboard navigate through the points using arrow keys and see a visible tooltip that shows the data:

https://github.com/user-attachments/assets/3861cc23-5f90-40ce-96d8-b41794a050dd

We're wondering if we could use a live region to announce the points as they are being navigated through and visually shown
. Ideally, there is a callback that triggers so we can make a live region announcement call in the same format as the string set on  `<path role="img" aria-label"`. Is there a callback we could rely on for this purpose?

**Comment by oysteinmoseng** [comment]
Hello @khiga8, great question! You have stumbled upon essentially one of the improvements we are working on for Highcharts accessibility. A chart with thousands of datapoints requires a different approach than one with 10.

Our solution to this will be based around a `role="application"` model, with announcements using ARIA live regions as you suggest. This seems to be the most robust and flexible way to handle large datasets, and has tested well with users so far. It's too early to share anything, but sometime mid 2025 we should be able to do this work for you.

To work around it until then, I would look at custom live region announcements as you suggest. Here is a starting point, hacking around built-in functionality to do most of the heavy lifting: https://jsfiddle.net/oysteinmoseng/sm09x3rh/1/ 
It is only briefly tested, and you will probably want to give the end-user some instructions for how to navigate too.

I would also consider sonification, which can be helpful for many users as a representation of large time series: https://www.highcharts.com/demo/highcharts/sonification-navigation.

Of course an alternative approach is to aggregate your data prior to visualizing it, for example using our data grouping functionality either on the [server](https://www.highcharts.com/docs/working-with-data/server-side-data-grouping) or [client side](https://www.highcharts.com/docs/stock/data-grouping).

I'll close this for now, since it is handled by ongoing efforts, but feel free to comment still.

**Comment by khiga8** [comment]
Thank you @oysteinmoseng for the detailed response on long-term plans as well as recommendations for how we can move forward  short-term 😄 .

**Comment by khiga8** [comment]
I started to implement the suggested monkey patch.

It seems to work quite well for the chart in the fiddle, but setting `pointDescriptionEnabledThreshold: 1` is resulting in problematic markup for other types of charts. I've logged this in:  [Keyboard focus on column chart with points above pointDescriptionEnabledThreshold results in errors #22199](https://github.com/highcharts/highcharts/issues/22199)! 🙇

**Comment by khiga8** [comment]
Hey @oysteinmoseng!

Your suggested approach of:
* setting `pointDescriptionEnabledThreshold: 1` (so that Highcharts doesn't expose any points)
* monkey patching the highlight function to make a live region call works

seems to work well for the chart example I linked in the issue! However, I'm wary that monkey patch will end up impacting all charts on the page, and it seems risky to have this be the default behavior for all charts. In some scenarios, [like the one I linked here](https://github.com/highcharts/highcharts/issues/22149#issuecomment-2491819813), I'd rather not set `pointDescriptionEnabledThreshold: 1` given the markup issues and given that Highcharts default accessibility behavior seems better.

I'm wondering if you have any suggestions for restricting this live region approach to a specific chart instance? I believe this might rely on being able to set some kind of event listener on a chart, rather than monkey patching the `highlight` function entirely.

**Comment by khiga8** [comment]
Nevermind, we were able to work this out by adding a conditional within the wrapper :) 

```
if (this.series.data.length > pointDescriptionEnabledThreshold) {
  // live region stuff
}
```
